crates/driver/src/tests/setup/blockchain.rs

```diff
@@ -97,14 +97,34 @@ impl QuotedOrder {
     pub fn buy_amount(&self) -> eth::U256 {
         match self.order.side {
             order::Side::Buy => self.buy,
-            order::Side::Sell => self.buy / self.order.surplus_factor,
+            order::Side::Sell => {
+                let fee = match &self.order.fee_policy {
+                    FeePolicy::PriceImprovement {
+                        factor: _,
+                        max_volume_factor: _,
+                        quote,
+                    } => quote.network_fee,
+                    _ => eth::U256::zero(),
+                };
+                (self.buy * (self.sell - fee) / self.sell) / self.order.surplus_factor
```

Comment:
Is it valid?

Reply:
I think we should always incorporate some network fee here (and not have an if branch just for PriceImprovement fees. That feels way too hacky.